internal/pkg/metadb/timestamps/uuid_test.go

```diff
@@ -18,87 +18,124 @@ import (
-	assert.Empty(t, ps, "LoadUUID should remove the UUID Property from the input array.")
-	assert.NoError(t, err, "LoadUUID should not return error for an empty string.")
+	t.Parallel()
 
-	// Returns the parsed UUID
-	anotherProperty := datastore.Property{Name: "another property", Value: "another value"}
-	u, ps, err = LoadUUID(
-		[]datastore.Property{
-			anotherProperty,
-			{
-				Name:  testName,
-				Value: testUUIDString,
+	testUUID := uuid.MustParse(testUUIDString)
+	testCases := []struct {
+		name      string
+		props     []datastore.Property
+		wantCode  codes.Code
+		wantProps []datastore.Property
+		wantUUID  uuid.UUID
+	}{
+		{"empty array", []datastore.Property{}, codes.Internal, []datastore.Property{}, uuid.Nil},
+		{
+			"empty string",
+			[]datastore.Property{
+				{
+					Name:  "empty string",
+					Value: "",
+				},
+			},
+			codes.OK,
+			[]datastore.Property{},
+			uuid.Nil,
+		},
+		{
+			"malformed string",
+			[]datastore.Property{
+				{
+					Name:  "malformed string",
+					Value: "this is not UUID",
+				},
+			},
+			codes.Unknown, // TODO(yuryu): Unknown is probably not appropriate
+			[]datastore.Property{},
+			uuid.Nil,
+		},
+		{
+			"valid",
+			[]datastore.Property{
+				{
+					Name:    "valid",
+					Value:   testUUIDString,
+					NoIndex: true,
+				},
+			},
+			codes.OK,
+			[]datastore.Property{},
+			testUUID,
+		},
+		{
+			"valid with extra properties",
+			[]datastore.Property{
+				{
+					Name:    "valid with extra properties",
+					Value:   testUUIDString,
+					NoIndex: true,
+				},
+				{
+					Name: "unrelated",
+				},
 			},
-		}, testName,
-	)
-	assert.Equal(t, uuid.MustParse(testUUIDString), u,
-		"LoadUUID should return a parsed UUID of the input string.")
-	if assert.Len(t, ps, 1, "LoadUUID should return the remaining properties.") {
-		assert.Equal(t, anotherProperty, ps[0], "LoadUUID should return the remaining properties.")
+			codes.OK,
+			[]datastore.Property{
+				{
+					Name: "unrelated",
+				},
+			},
+			testUUID,
+		},
 	}
-	assert.NoError(t, err, "LoadUUID should not return error when successful.")
-}
 
-func TestUUID_LoadUUIDMalformedString(t *testing.T) {
-	u, ps, err := LoadUUID(
-		[]datastore.Property{
-			{
-				Name:  testName,
-				Value: "not a UUID",
-			},
-		}, testName,
-	)
-	assert.Equal(t, uuid.Nil, u, "LoadUUID should return uuid.Nil in case of parse errors.")
-	assert.Empty(t, ps, "LoadUUID should still remove the UUID property in case of parse errors.")
-	assert.Error(t, err, "LoadUUID should return a non-nil err in case of parse errors.")
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			uuid, ps, err := LoadUUID(tc.props, tc.name)
+			if got := status.Code(err); got != tc.wantCode {
+				t.Errorf("LoadUUID(): returned status code %v, want %v", got, tc.wantCode)
+			}
+			if !cmp.Equal(tc.wantUUID, uuid) {
+				t.Errorf("LoadUUID() uuid = want %v, got %v", tc.wantUUID, uuid)
```

Comment:
Prefer order of actual (got) and then expected (want) rather than other way around

https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures